### PR TITLE
Ability to use UUID Class and generic classes that support fromString method

### PR DESCRIPTION
**Pull request by danielgruber**
What do you think? is this a nice extension to the existing helpers?

### CODE REVIEW

**Comment by zifnab87** [comment]
@danielgruber Hi thanks for the contribution. Could you change the entities or create new ones if needed and create passing test(s) that show what clearly what is supported? Thanks

**Comment by danielgruber** [comment]
Hi @zifnab87 Thanks for your feedback and sorry for the delay :) I additionally fixed the ability to use foreign keys and added tests. Can you check for that?

**Comment by AlbericTrancart** [comment]
:+1: :+1: :+1: Would be very happy to have this

File: src/main/java/springboot/rest/specifications/CustomSpecifications.java [hunk 1]
```diff
@@ -199,28 +199,58 @@ public Predicate searchInAllAttributesPredicate(CriteriaBuilder builder, Root ro
...
-            return builder.equal(root.get(a.getName()), val);
-        } else if (a.isAssociation()) {
-            return prepareJoinAssociatedPredicate(root, a, val);
-        } else if(isSerializable(a)) {
-            try {
-                return builder.equal (root.get(a.getName()), a.getJavaType().getConstructors()[0].newInstance(val));
-            } catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
-                throw new IllegalArgumentException("equality/inequality is currently supported on primitives, enums and serializables with string constructor", e);
+        }
+        else if (join == null) {
```
Comment (zifnab87) [inline src/main/java/springboot/rest/specifications/CustomSpecifications.java]: this part is really complicated after the copy paste for join  == null or join != null

File: src/main/java/springboot/rest/specifications/CustomSpecifications.java [hunk 2]
```diff
@@ -216,14 +228,26 @@ else if (join == null) {
...
             }
             else if (a.isAssociation()) {
-                return prepareJoinAssociatedPredicate(root, a, val);
+                return prepareJoinAssociatedPredicate(builder, root, a, val);
+            } else if(isSerializableAndFromString(a)) {
+                try {
+                    return builder.equal (root.get(a.getName()), a.getJavaType().getMethod("fromString", String.class).invoke(null, val.toString()));
+                } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+                    throw new IllegalArgumentException("equality/inequality is currently supported on primitives, enums and serializables with string constructor", e);
+                }
```
Comment (zifnab87) [inline src/main/java/springboot/rest/specifications/CustomSpecifications.java]: those are identical need to be extracted

File: src/main/java/springboot/rest/specifications/CustomSpecifications.java [hunk 3]
```diff
@@ -216,14 +228,26 @@ else if (join == null) {
...
         else if (join != null) {
             if (isEnum(a)) {
                 return builder.equal(join.get(a.getName()), Enum.valueOf(Class.class.cast(a.getJavaType()), (String) val));
             } else if (isPrimitive(a)) {
                 return builder.equal(join.get(a.getName()), val);
+            } else if(isSerializableAndFromString(a)) {
+                try {
+                    return builder.equal (join.get(a.getName()), a.getJavaType().getMethod("fromString", String.class).invoke(null, val.toString()));
+                } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+                    throw new IllegalArgumentException("equality/inequality is currently supported on primitives, enums and serializables with string constructor", e);
```
Comment (zifnab87) [inline src/main/java/springboot/rest/specifications/CustomSpecifications.java]: those are identical need to be extracted

File: src/main/java/springboot/rest/specifications/CustomSpecifications.java (hunk 2 above)
Comment (danielgruber) [reply src/main/java/springboot/rest/specifications/CustomSpecifications.java]: Please recheck :)

File: src/main/java/springboot/rest/specifications/CustomSpecifications.java (hunk 3 above)
Comment (danielgruber) [reply src/main/java/springboot/rest/specifications/CustomSpecifications.java]: Please recheck :)

**Comment by zifnab87** [comment]
this was implemented in a much simpler way through #7 which made less code changes and made those two tests more robust. Thank you for the patience and effort! :)

**Pull request closed without merging**